vm/src/byteslike.rs

```diff
@@ -35,6 +52,44 @@ impl PyBytesLike {
         match self {
             PyBytesLike::Bytes(b) => f(b.get_value()),
             PyBytesLike::Bytearray(b) => f(&b.borrow_value().elements),
+            PyBytesLike::Array(array) => f(&*array.get_bytes()),
+        }
+    }
+}
+
+pub enum PyBuffer {
```

Comment:
Oh, buffer protocol is splited from `PyBytesLike`, right?

Reply:
They are pretty much the same except `PyBytesLike` is read only. Looking at https://docs.python.org/3/glossary.html#term-bytes-like-object, this probably should be named `PyRWBytesLike` instead.